src/util/preferredLanguage.js

```diff
@@ -0,0 +1,15 @@
+import { hasExtension, getExtension } from "./extensions";
+
+export const getPreferredLanguage = hasExtension("getPreferredLanguage") ?
+  getExtension("getPreferredLanguage")
+  : () => {
+  let browserLang = window.navigator.language.split("-")[0],
+      narrativeLang = "",
+      urlParts = window.location.pathname.toLowerCase().split("/");
+
+  if (urlParts.length > 0 && urlParts[1] === "narratives") {
+      narrativeLang = urlParts[urlParts.length - 2];
```

Comment:
This is also a bit of logic specific to our 2019-nCoV narratives which doesn't have any fundamental basis in Auspice's view of things. I think we'd want to keep that separation in the default implementation of this function.